xmtp_mls/src/groups/mod.rs

```diff
@@ -187,6 +187,8 @@ pub enum GroupError {
     SqlKeyStore(#[from] sql_key_store::SqlKeyStoreError),
     #[error("No pending commit found")]
     MissingPendingCommit,
+    #[error("Sync failed to wait for intent: {0}")]
+    SyncFailedToWait(String),
```

Comment:
@neekolas adding this so it's easier to catch the error that is causing group creates to fail.